packages/device-connector/src/arduino-cli/client.ts

```diff
@@ -48,20 +45,24 @@ export class RPCClient {
       oneofs: true
     })
 
+    const options: ChannelOptions = {
+      'grpc.keepalive_permit_without_calls': 1
+    }
+
     const deadline = new Date()
     deadline.setSeconds(deadline.getSeconds() + 5)
 
-    return await new Promise((resolve, reject) => {
+    return await new Promise((resolve) => {
       const arduinoGrpcObject = grpc.loadPackageDefinition(loadedProto) as unknown as ArduinoProtoGrpcType
 
-      const arduinoServiceClient = new arduinoGrpcObject.cc.arduino.cli.commands.v1.ArduinoCoreService(this.address, grpc.credentials.createInsecure())
+      const arduinoServiceClient = new arduinoGrpcObject.cc.arduino.cli.commands.v1.ArduinoCoreService(this.address, grpc.credentials.createInsecure(), options)
       arduinoServiceClient.waitForReady(deadline, (error: Error | undefined) => {
-        if (error != null) {
-          return reject(new Error(error.message))
+        if (error !== undefined) {
```

Comment:
Not sure how this is handled, but convention is err is null if everything is fine.
`err != null` handles `undefined` aswell.

Reply:
The handling is a bit weird here. This needs to be reworked slightly later on. For now it's changed to catch any null error.